workspaces/agent-forge/plugins/agent-forge/src/a2a/client.ts

```diff
@@ -0,0 +1,562 @@
+      if (error instanceof RpcError) {
+        throw error;
+      } else {
+        throw new RpcError(
+          -32603, // Use literal value for ErrorCodeInternalError
+          `Failed to process response: ${
+            error instanceof Error ? error.message : String(error)
+          }`,
+          error,
+        );
+      }
+    }
+  }
+
+  /**
+   * Handles streaming Server-Sent Events (SSE) responses.
+   * Returns an AsyncIterable that yields ONLY the 'result' payloads (events).
+   * Throws RpcError if an error is received in the stream.
+   */
+  private async *_handleStreamingResponse<StreamRes extends JSONRPCResponse>( // Takes full Response type
+    response: Response,
+    expectedMethod?: string, // Optional: helps in debugging
+  ): AsyncIterable<StreamRes['result']> {
+    // Yield type is now the 'result' property of StreamRes
+    if (!response.ok || !response.body) {
+      let errorText: string | null = null;
+      try {
+        errorText = await response.text();
+      } catch (_) {
+        /* Ignore read error */
+      }
+      //console.error(
+      //  `HTTP error ${response.status} received for streaming method ${
+      //    expectedMethod || "unknown"
+      //  }.`,
+      //  errorText ? `Response: ${errorText}` : ""
+      //);
+      throw new Error(
+        `HTTP error ${response.status}: ${response.statusText} - Failed to establish stream.`,
+      );
+    }
+
+    const reader = response.body
+      .pipeThrough(new TextDecoderStream())
+      .getReader();
+    let buffer = '';
+
+    try {
+      while (true) {
+        const { done, value } = await reader.read();
+
+        if (done) {
+          // Process any remaining data in the buffer before exiting
+          if (buffer.trim()) {
+            //console.warn(
+            //  `SSE stream ended with partial data in buffer for method ${expectedMethod}: ${buffer}`
+            //);
+          }
+          break;
+        }
+
+        buffer += value;
+        const lines = buffer.replace(/\r/g, '').split('\n\n'); // SSE messages end with \n\n
+        buffer = lines.pop() || ''; // Keep potential partial message
+        for (const message of lines) {
+          if (message.startsWith('data: ')) {
+            const dataLine = message.substring('data: '.length).trim();
+            if (dataLine) {
+              // Ensure data is not empty
+              try {
+                // Parse as the specific JSONRPCResponse type StreamRes
+                const parsedData = JSON.parse(dataLine) as StreamRes;
+                // Basic validation of streamed data structure
+                if (
+                  typeof parsedData !== 'object' ||
+                  parsedData === null ||
+                  !('jsonrpc' in parsedData && parsedData.jsonrpc === '2.0')
+                ) {
+                  //console.error(
+                  //  `Invalid SSE data structure received for method ${expectedMethod}:`,
```

Comment:
Let's just removed the commented out code in all this file, sorry, don't want to spam you with the same comment.